kratos/geometries/quadrilateral_2d_4.h

```diff
@@ -1040,6 +1066,210 @@ template<class TPointType> class Quadrilateral2D4
         return shape_functions_local_gradients;
     }
 
+
+    inline bool QuadBoxOverlap(Point<3, double>& boxcenter, Point<3, double>& boxhalfsize, std::vector< Point<3, double> >& quadverts)
+    {
+        /*    use separating axis theorem to test overlap between quadrilateral and box  */
+        /*    need to test for overlap in these directions:                              */
+        /*    1) the {x,y,z}-directions                                                  */
+        /*    2) normal of the quadrilateral                                             */
+        /*    3) crossproduct(edge from quad, {x,y,z}-directin)                          */
+        /*       this gives 4x3=12 more tests                                            */
+        
+        double min,max,d,fex,fey,fez;
+        array_1d<double,3 > v0,v1,v2,v3;
+        array_1d<double,3 > axis;
+        array_1d<double,3 > normal,e0,e1,e2,e3;
+        
+        /* This is the fastest branch on Sun */
+        /* move everything so that the boxcenter is in (0,0,0) */
+        noalias(v0) = quadverts[0] - boxcenter;
+        noalias(v1) = quadverts[1] - boxcenter;
+        noalias(v2) = quadverts[2] - boxcenter;
+        noalias(v3) = quadverts[3] - boxcenter;
+        
+        /* compute quadrilateral edges */
+        noalias(e0) = v1 - v0;         /* quad edge 0 */
+        noalias(e1) = v2 - v1;         /* quad edge 1 */
+        noalias(e2) = v3 - v2;         /* quad edge 2 */
+        noalias(e3) = v0 - v3;         /* quad edge 3 */
+        
+        /* Bullet 3:  */
+        /* test the 12 tests first (this was faster) */
+        fex = fabs(e0[0]);
```

Comment:
I reccomend to use `std::abs` instead of fabs: https://stackoverflow.com/questions/33738509/whats-the-difference-between-abs-vs-fabs-in-c